core/services/keeper/orm.go

```diff
@@ -134,19 +134,27 @@ func (korm ORM) SetLastRunHeightForUpkeepOnJob(db *gorm.DB, jobID int32, upkeepI
 		).Error
 }
 
-func (korm ORM) CreateEthTransactionForUpkeep(sqlDB *sql.DB, upkeep UpkeepRegistration, payload []byte, maxUnconfirmedTXs uint64) (models.EthTx, error) {
+func (korm ORM) CreateEthTransactionForUpkeep(tx *gorm.DB, upkeep UpkeepRegistration, payload []byte, maxUnconfirmedTXs uint64) (models.EthTx, error) {
 	var etx models.EthTx
 	ctx, cancel := postgres.DefaultQueryCtx()
 	defer cancel()
 
+	sqlTx, ok := tx.ConnPool.(*sql.Tx)
+	if !ok {
+		return etx, errors.New("unable to get tx from conn pool")
+	}
+	sqlDB, err := tx.DB()
+	if err != nil {
+		return etx, err
+	}
 	from := upkeep.Registry.FromAddress.Address()
-	err := utils.CheckOKToTransmit(ctx, sqlDB, from, maxUnconfirmedTXs)
+	err = utils.CheckOKToTransmit(ctx, sqlDB, from, maxUnconfirmedTXs)
 	if err != nil {
 		return etx, errors.Wrap(err, "transmitter#CreateEthTransaction")
 	}
 
 	value := 0
-	err = sqlDB.QueryRowContext(ctx, `
+	err = sqlTx.QueryRowContext(ctx, `
```

Comment:
sqlDB is importantly NOT the transaction - chatted with jinzhu about it

Reply:
I'll do a scan and fix elsewhere in the codebase in a follow up PR

Reply:
That is a nasty surprise.

We should fix in this PR imo

Reply:
Did a scan, didn't find any other places that explicitly needed a tx